### DISCUSSION TITLE
Module not found: Can't resolve '@radix-ui/react-select'

### DISCUSSION THREAD

**Discussion by turnsy** [shadcn-ui/ui#4784] [General]
I added the "Select" component using the exact command from the shadcn website, but keep getting this error:
`Module not found: Can't resolve '@radix-ui/react-select'`, which is imported from the shadcn Select component for primitives.

```ts
// select.tsx
import * as SelectPrimitive from "@radix-ui/react-select";
```

I am able to import other `@radix-ui` packages, for example `@radix-ui/accordion`, and I have confirmed that`@radix-ui/react-select` exists in the correct directory adjacent to `@radix-ui/accordion`. This confirms to me it isn't a global configuration problem

Things I have tried:
- Reinstalling the `@radix-ui/react-select` package
- Reinstalling ALL packages
- Clearing npm cache
- Explicit importing ex. "`../../node_modules/...etc`"
- Switching to yarn from npm
- Importing components directly from `@radix-ui/react-select`, this also does not work.
- Forcing an alias in `next.config.mjs` and `tsconfig.json`
- Using `import * as SelectPrimitive from '@radix-ui/react-select/dist/index';`
- Any other common solution that Claude AI suggested

What could I be missing here? This blocks the usage of the Select component, and im curious if anyone else has seen this problem before.

**Comment by Shriharsh-Deshmukh** [comment]
You're likely dealing with a bundler or module resolution issue. Here's how to address it:

1. **Ensure Correct Installation:** Verify that `@radix-ui/react-select` is properly installed by checking your `package.json` and running:
   ```bash
   npm ls @radix-ui/react-select
   ```

2. **Manual Path Resolution:** If the package exists, but still isn't resolving, try importing directly from the package's `index` file:
   ```typescript
   import * as SelectPrimitive from '@radix-ui/react-select/dist/index';
   ```

3. **Next.js Module Resolution:** Ensure your `next.config.js` includes proper module resolution:
   ```javascript
   const path = require('path');
   module.exports = {
     webpack: (config) => {
       config.resolve.alias['@radix-ui/react-select'] = path.resolve(__dirname, 'node_modules/@radix-ui/react-select');
       return config;
     },
   };
   ```

4. **Check for Peer Dependency Conflicts:** Ensure there are no peer dependency conflicts by checking your `node_modules` for duplicate Radix packages.

This should resolve the import issue.

**Reply by turnsy** [reply]
I tried steps 1-3. Do you mind explaining 4 further?

**Comment by uvislive** [comment]
i am also facing same issue, tried all solutions but did not work

**Comment by Aribodo** [comment]
Im having a similar issue with the @radix-ui/react-tabs. No other issue with my other radix imports.

**Reply by Aribodo** [reply]
Update:
It was some kind of cache issue with my docker environment. I just had to clear it.

**Comment by IvanTsxx** [comment]
is it possible that the error is due to next js 15 and react 19?

**Reply by IvanTsxx** [reply]
this problem is provacte for turbopack

**Comment by viveleroi** [comment]
I'm having this same problem with a fresh install of https://github.com/vercel/nextjs-postgres-nextauth-tailwindcss-template

None of these solutions appear to fix it.

**Reply by IvanTsxx** [reply]
The solution for me was to run pnpm dev without turbopack

**Comment by DanUgbeye** [comment]
It appears that Next 15 and react 19-rc with turbopack is broken for @radix-ui/react-collapsible and it affected @radix-ui/react-roving-focus, @radix-ui/react-focus-scope, @radix-ui/react-dropdown-menu, @radix-ui/react-dismissable-layer

Had to turn off turbo pack to fix

**Reply by gioruu** [reply]
Same boat brother...

**Reply by gioruu** [reply]
Update:
Found a solution.
Delete node_modules + pnpm-lock and REINSTALL EVERYTHING IN NPM. It was a **pnpm problem**, can't believe it.. took me two freaking days of headaches trying everything.

**Reply by DanUgbeye** [reply]
Many thanks to you or finding this fix, I'll try it out

**Reply by tliam1** [reply]
This worked for me as well. Moreover, after installing with npm and removing package-lock.json, using PNPM worked fine.

**Reply by Martin-K-Kamir** [reply]
Yup, worked for me too. I changed from pnpm to npm for the installation. I had issues when I added the shadcnui Accordion and Avatar components.

**Reply by pantharshit007** [reply]
> Moreover, after installing with npm and removing package-lock.j

That didn't work for me, when using npm or not using turbo is the solution :/
![image](https://github.com/user-attachments/assets/d90c3417-4748-452a-8626-f8f9a827c65a)

**Comment by getorca** [comment]
downgrading from node 23 to node 22lts resolved similar issues i was having
